### DISCUSSION TITLE
Strange behaviour with ingest files

### DISCUSSION THREAD

**Discussion by Sahale** [Security-Onion-Solutions/securityonion#4888] [Unsupported Versions]
Hello!
I've created some ingests files in /opt/so/saltstack/local/salt/elasticsearch/files/ingest/ for tests. After restarting via so-elasticsearch-restart they successfully added to elastic. After that I deleted these old test files, deleted pipelines from elastic, added new ingest file to custom ingest parsers folder, restarted elastic one more time and but all deleted ingests returned back.
Is there any way to delete unneeded pipelines? Thank you!

**Comment by gebhard73** [comment]
Have a look at `/opt/so/conf/elasticsearch/ingest` - as far as I can remember custom files aren't deleted automatically here.

**Reply by Sahale** [reply]
Thanks for reply!
I've cleaned this folder on every node, restarted elastic with so-elasticsearch-restart and nothing happened, I can see deleted pipelines with so-elasticsearch-pipelines-list.

**Reply by gebhard73** [reply]
AFAIK only _custom stuff_ should be deleted from this folder. What is the status of you installation - does it work as expected or not? If not you should re-create the content of the folder with the standard stuff (saltstack should handle this).

**Reply by Sahale** [reply]
Yes, I deleted only custom stuff, I use names mask, so it's easy to find all non-default files.
Status is OK, all nodes work fine except this restoration of deleted ingests.

**Reply by gebhard73** [reply]
So the deleted ingest still appear? Can you show a potentially redacted example?

**Reply by Sahale** [reply]
After some cleaning of folders and restarts of elastic on manager node, deleted ingests disappeared from it. Strange. Now I can see this:
Manager node: 
```
# ls -lah /opt/so/saltstack/local/salt/elasticsearch/files/ingest/
total 12K
drwxr-xr-x. 2 socore socore 4.0K Jul 22 15:57 .
drwxr-xr-x. 4 socore socore 4.0K Jul  5 09:35 ..
-rw-r--r--. 1 socore socore 3.6K Jul 22 15:57 X-mikrotik

# ls -lah /opt/so/conf/elasticsearch/ingest/
total 336K
drwxr-xr-x. 2 elasticsearch socore 4.0K Jul 22 15:58 .
drwxr-xr-x. 4 elasticsearch socore 4.0K Jul  5 09:43 ..
-rw-r--r--. 1 elasticsearch socore 3.6K Jul 22 15:58 X-mikrotik
-rw-r--r--. 1 elasticsearch socore  352 Jun 22 10:52 beats.common
-rw-r--r--. 1 elasticsearch socore 3.1K Jun 22 10:52 common
-rw-r--r--. 1 elasticsearch socore 2.1K Jun 22 10:52 common.nids
-rw-r--r--. 1 elasticsearch socore 1.6K Jun 22 10:52 dns.tld
-rw-r--r--. 1 elasticsearch socore 3.0K Jul  5 09:36 filterlog
.....
-rw-r--r--. 1 elasticsearch socore 5.0K Jun 22 10:52 zeek.x509

#  so-elasticsearch-pipelines-list 
[
  "X-mikrotik",
  "beats.common",
  "common",
  "common.nids",
  "dns.tld",
  .....
  "zeek.x509"
]
```

Heavy node:
```
# ls -lah /opt/so/conf/elasticsearch/ingest/
total 336K
drwxr-xr-x. 2 elasticsearch socore 4.0K Jul 23 09:47 .
drwxr-xr-x. 4 elasticsearch socore 4.0K Jul  5 09:51 ..
-rw-r--r--. 1 elasticsearch socore 3.6K Jul 22 16:00 X-mikrotik
-rw-r--r--. 1 elasticsearch socore  352 Jul  1 15:24 beats.common
-rw-r--r--. 1 elasticsearch socore 3.1K Jul  1 15:24 common
-rw-r--r--. 1 elasticsearch socore 2.1K Jul  1 15:24 common.nids
-rw-r--r--. 1 elasticsearch socore 1.6K Jul  1 15:24 dns.tld
-rw-r--r--. 1 elasticsearch socore 3.0K Jul  5 09:51 filterlog
......
-rw-r--r--. 1 elasticsearch socore 5.0K Jul  1 15:24 zeek.x509


# so-elasticsearch-pipelines-list 
[
  "X-mikrotik",
  "Xmikrotik",
  "Xmikrotiktest",
  "beats.common",
  "common",
  "common.nids",
  "dns.tld",
.....
  "zeek.x509"
]
```

Elastic on heavy node was restarted too, nothing changed.

**Reply by gebhard73** [reply]
This could be because of salt refreshing the config every 15 minutes .... so everything is now sorted out?

**Reply by Sahale** [reply]
Well, no.
I'm trying to find out what I have to delete on nodes. I deleted custom files from /var/cache/salt/minion/files/base/elasticsearch/files/ingest/ on manager and from /opt/so/conf/elasticsearch/ingest/ and /var/cache/salt/minion/files/base/elasticsearch/files/ingest/ on heavy nodes, than rebooted elastic, but pipelines still in the list. Tried to find test files with mlocate on manager and heavy nodes, but there is no one.

**Reply by gebhard73** [reply]
from my understanding the only relevant directory should be `/opt/so/saltstack/local/salt/elasticsearch/files/ingest/` on the manager
